library/ssl_cli.c

```diff
@@ -2235,20 +2244,23 @@ static int ssl_parse_server_hello( mbedtls_ssl_context *ssl )
      */
     comp = buf[37 + n];
 
+    int bad_comp = 0;
```

Comment:
Non-blocker: this screams for an auxiliary function `is_compression_ok`.

Even without an auxiliary function, it would be easier to follow the code with a single variable `accept_comp`, extended to be defined (but always 0) even without `MBEDTLS_ZLIB_SUPPORT`.